### PR TITLE
Name the folder causing a repository error

### LINKED ISSUE #1204: Improve error message when push to hub fails because it is not a git repo
**Is your feature request related to a problem? Please describe.**

I can't figure out which folder I need to modify to get a `push_to_hub` to work. I would like to improve the already excellent error message:

```python
/usr/lib/python3.8/site-packages/huggingface_hub/repository.py in clone_from(self, repo_url, token)
    760                 # Check if the folder is the root of a git repository
    761                 if not is_git_repo(self.local_dir):
--> 762                     raise EnvironmentError(
    763                         "Tried to clone a repository in a non-empty folder that isn't a"
    764                         " git repository. If you really want to do this, do it"

OSError: Tried to clone a repository in a non-empty folder that isn't a git repository. If you really want to do this, do it manually:
git init && git remote add origin && git pull origin main
 or clone repo to a new folder and move your existing files there afterwards.
OSError: Tried to clone a repository in a non-empty folder that isn't a git repository. If you really want to do this, do it manually:
git init && git remote add origin && git pull origin main
 or clone repo to a new folder and move your existing files there afterwards.
```

**Describe the solution you'd like**

I would like the `self.local_dir` variable to be interpolated into the error message: 

```
"Tried to clone a repository in a non-empty folder that isn't a"
f" git repository ('{self.local_dir}'). If you really want to do this, do it"
```

**Describe alternatives you've considered**

None

**Additional context**

 None

### PR DESCRIPTION
**Pull request by payoto**
Improve the error message when a repository is being created in a non-empty folder by saying which folder is causing the error.

Fixes #1204

### CODE REVIEW

**Review by Wauplin — APPROVED** [review]
Thanks for making the change @payoto !

**Comment by osanseviero** [comment]
@Wauplin FYI external members don't have write access to the repo, so you will need to merge yourself once you think it's ready

File: src/huggingface_hub/repository.py
Comment (Wauplin) [inline src/huggingface_hub/repository.py:768]: Code quality:

```suggestion
                    raise EnvironmentError(
                        "Tried to clone a repository in a non-empty folder that isn't"
                        f" a git repository ('{self.local_dir}'). If you really want to"
                        f" do this, do it manually:\n cd {self.local_dir} && git init"
                        " && git remote add origin && git pull origin main\n or clone"
                        " repo to a new folder and move your existing files there"
                        " afterwards."
                    )
```

File: src/huggingface_hub/repository.py
Comment (Wauplin) [inline src/huggingface_hub/repository.py:767]:
```suggestion
                        "Tried to clone a repository in a non-empty folder that isn't"
                        f" a git repository ('{self.local_dir}'). If you really want to"
                        f" do this, do it manually:\n cd {self.local_dir} && git init"
                        " && git remote add origin && git pull origin main\n or clone"
                        " repo to a new folder and move your existing files there"
                        " afterwards."
```

**Comment by Wauplin** [comment]
@payoto Could you run `make style` to apply code quality rules please ? Or just commit the suggestion I made. I can't push on your branch so the change has to come from you. I'll merge the PR just after :)

File: src/huggingface_hub/repository.py
Comment (payoto) [reply src/huggingface_hub/repository.py]: Yep, done

**Comment by payoto** [comment]
@Wauplin no problem, done :)

**Comment by payoto** [comment]
FYI, just noticed that if you get maintainer status on the repo you can do edits to 3rd party branches:

Screenshot 2022-11-21 at 09 40 24

**Comment by Wauplin** [comment]
Thanks @payoto !

**Pull request merged**
